src/Microsoft.Extensions.Configuration/ConfigurationBuilderExtensions.cs

```diff
@@ -7,9 +7,42 @@
 
 namespace Microsoft.Extensions.Configuration
 {
-    public static class MemoryConfigurationExtensions
+    public static class ConfigurationBuilderExtensions
     {
         /// <summary>
+        /// Includes an existing IConfiguration as a configuration provider to <paramref name="configuraton"/>.
+        /// </summary>
+        /// <param name="configurationBuilder">The <see cref="IConfigurationBuilder"/> to add to.</param>
+        /// <param name="config">The <see cref="IConfiguration"/> to include.</param>
+        /// <returns>The <see cref="IConfigurationBuilder"/>.</returns>
+        public static IConfigurationBuilder AddConfiguration(this IConfigurationBuilder configurationBuilder, IConfiguration configuration)
```

Comment:
Since we are going away from the other extensions naming pattern.  How about something like Copy/Include/Import depending on exactly what kind of intent we want to imply.